src/utils/characterSet.ts

```diff
@@ -0,0 +1,24 @@
+import emojiRegex                     from "emoji-regex";
+import { characterSet, CharacterSet } from "../base";
+
+
+export function findCharacterSet(name: string): CharacterSet {
+    const label = name.substring(0, name.lastIndexOf('.'));
```

Comment:
This will return everything except the TLD, not just the first label.

Reply:
right! it must be `indexOf`, updating.